### PR TITLE
Assignments/game of life

### PR DESCRIPTION
**Pull request by EllisWaterman**

### CODE REVIEW

File: assignments/game-of-life/code.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const CELLSIZE = .1;
+const rows = Math.floor(height / CELLSIZE)
+const cols = Math.floor(width / CELLSIZE)
+
+const cellArray =
+  Array(rows).fill().map(() =>
+    Array(cols).fill().map(() => ({ health: 'dead' })));
```
Comment (gigamonkey) [inline assignments/game-of-life/code.js:7]: While this is a correct use of objects, in this case, where every cell is either alive or dead, I'd probably just use booleans. I.e. set every cell to `false` initially and then switch it to `true` when it is alive. That will simplify the boolean expressions in `doTheLive`.

**Review by gigamonkey — CHANGES_REQUESTED** [review]
Looking good but I think the logic isn't quite right.

**Comment by EllisWaterman** [comment]
I made it work. I looked up delay and switch break statement.

File: assignments/game-of-life/code.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        }
+      }
+    }
+  }
+}
+
+const countLivingNeighbors = (locations, i, j) => {
+  let livingthings = 0
+  locations.forEach(function (location) {
+    switch (location) {
```
Comment (gigamonkey) [inline assignments/game-of-life/code.js:68]: This is a fine way to do it though I suspect there are some underlying mathematical regularities that you could take advantage of. (Though one way to get there might require knowing trig.)

File: assignments/game-of-life/code.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  return locations;
+}
+
+const go = () => {
+  doTheyLive()
+  drawNext()
+  current = next
+  next = Array(rows).fill().map(() =>
+    Array(cols).fill().map(() => (false)));
+  setTimeout(go, 100);
```
Comment (gigamonkey) [inline assignments/game-of-life/code.js:145]: Glad you figured this out. Though this framework is set up for animation so you don't actually need to do things this way, you just need to write a function that you can pass to `animate`. But in some ways it's more impressive that you figured this out on your own, presumably from Googling and reading docs. So nice work.

**Review by gigamonkey — APPROVED** [review]
This is very impressive work. There are ways to simplify the code that I may make a video about but everything is very clean and makes sense.

**Pull request merged**